### ISSUE TITLE
Configure HttpClient

### ISSUE THREAD

**Issue by Leptopoda** [nextcloud/neon#788]
- add  follow redirects (discuss the security implications!)
- add keepalive (persistent connection)

**Comment by Leptopoda** [comment]
Looks like the `persistentConnection` is only a header indicating whether a persistent connection is kept.
The property to change would be `idleTimeout` wich already defaults to a reasonable 15 seconds.

I think this is enough but @provokateurin do cou think we should raise it nevertheless? 


https://api.flutter.dev/flutter/dart-io/HttpClient/idleTimeout.html
https://api.flutter.dev/flutter/dart-io/HttpHeaders/persistentConnection.html

**Comment by provokateurin** [comment]
I think that is enough, we also don't want people to wait forever while the network is not working at all.

The other points should still be investigated.

**Comment by provokateurin** [comment]
Shall we switch to cronet_http_embedded now?

**Comment by Leptopoda** [comment]
In theory yes but we should not rush it.

1. the embedded version is probably going to be deprecated in the long run while the normal version is going to take a compile time option to distinguish between both. Not a stopper for us as migration would be straight forward.

2. With cronet using ffi to communicate to the native lib we loose the benefits of streams. While they expose streaming interfaces all calls are collected into a single buffer copied over to the native code and then sent (similar for receiving). This would not be a big issue for the api requests but with large webdav files we might hit memory limitations. 
Don't quote me on this point as I haven't had the time to dig into this deeper and do actual tests and I might have misunderstood some code I read  a few weeks ago.

**Comment by provokateurin** [comment]
Oh, no streams support is kind of a deal breaker. Users should be able to download a 4GB file on a device with only 2GB RAM.

**Comment by Leptopoda** [comment]
to my knowledge webdav isn't really benefiting from http2 or http3 as there probably aren't many concurrent dowloads ever happening.
We could just use dart:io client for webdav.

**Comment by Leptopoda** [comment]
Moving to the neon_framework as dynamite should accept any client

**Comment by provokateurin** [comment]
Anything that still needs to be done?

**Comment by Leptopoda** [comment]
I still want to use a client that supports http2 or even http3.
On Android my favorite would be the `cronet` client, but that either relies on the play services or bundling it ourselves would increase the apk size by roughly 14MB.
A different approach would be the `ok_http` client. It only supports http2 but doesn't bloat the apk size (https://github.com/dart-lang/http/tree/master/pkgs/ok_http).

**Comment by provokateurin** [comment]
Yeah GPlay or 14 MB are not great options, let's do ok_http then.
How about closing this issue and just creating a new one for ok_http to decrease the overall scope?
